src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -818,6 +823,11 @@ private bool InferNamedPipesInformation()
                         return false;
                     }
 
+                    if(tokensByBackSlash[4].StartsWith(NamedPipeInstanceNameHeader))
```

Comment:
@geleems the pipe instance name computation has many more conditions. I don't think you should tackle it as part of this PR